### ISSUE TITLE
meta(`1.0`): tracking issue for 1.0 milestone

### ISSUE THREAD

**Issue by zerosnacks** [foundry-rs/foundry#8574]
Tracking issue for `1.0` milestone

Ticket to be used for discussion and track high-level progress.

Link to trackers: 

- `foundry-rs/foundry`: https://github.com/foundry-rs/foundry/milestone/1
- `foundry-rs/compilers`: https://github.com/foundry-rs/compilers/milestone/1
- `foundry-rs/block-explorers`: https://github.com/foundry-rs/block-explorers/milestone/1
- `foundry-rs/foundry-fork-db`: https://github.com/foundry-rs/foundry-fork-db/milestone/1
- `foundry-rs/book`: https://github.com/foundry-rs/book/milestone/1

To discuss:
- Maintenance status of https://github.com/foundry-rs/hardhat

Tip:

Use the issue tracker to more easily view / edit the milestone: https://github.com/foundry-rs/foundry/issues?q=is%3Aopen+is%3Aissue+milestone%3Av1.0.0

------

Preparation: 

- [x] Identify all active tickets
- [x] Label all tickets + PRs
- [x] Apply a high-level filter to milestone tickets, make sure all tickets in milestone are actionable
- [x] Determine blockers / ticket dependency tree
- [x] Label all tickets with a priority according to bug severity / feature impact / complexity matrix
- [x] Determine status of current open PRs

------

Abstract scope `1.0`:

- Focus on stability, correctness and completeness:
  - Testing (unit, fuzz, invariant)
  - Contract deployment
  - Contract verification
  - Gas measurement / reporting
  - Test coverage
- Compatibility with EVM / Ethereum L2s (precompiles)
- Priority: `forge` > `anvil` + `cast` + `debugger`> `chisel`
- Correct behavior of `--json` flag and correct rendering to `stdout` / `stderr`

------

Breaking changes (https://github.com/foundry-rs/foundry/labels/T-likely-breaking) - to be done before 1.0 release, with the possibility to opt out:

- [x] https://github.com/foundry-rs/foundry/issues/3437
- [x] https://github.com/foundry-rs/foundry/issues/2486
- [x] https://github.com/foundry-rs/foundry/issues/4437
- [x] https://github.com/foundry-rs/foundry/pull/9412
- [x] https://github.com/foundry-rs/foundry/issues/8931
- [x] https://github.com/foundry-rs/foundry/pull/8910
- [ ] https://github.com/foundry-rs/foundry/pull/9522

------

Out of scope for `1.0`, better suited after:

- Extension roadmap: https://github.com/foundry-rs/foundry/issues/8266 + https://github.com/foundry-rs/foundry/labels/A-extensions
- Private transaction support (bundles): https://github.com/foundry-rs/foundry/labels/A-private-transactions
- Additional testing methodologies: https://github.com/foundry-rs/foundry/issues/8267
- Improve use of Foundry as a library (`forge-builder`)
- Rewrite of `anvil` to build on top of `reth` (blocked by `reth-sdk`)
- Rewrite of the formatter and https://github.com/foundry-rs/foundry/labels/Cmd-forge-fmt tickets
- Addition of `forge lint` / static analysis
- Rewrite of CLI + extendible `foundry.toml`: https://github.com/foundry-rs/foundry/issues/7322
- Integration of forked version of Foundry: https://github.com/foundry-rs/foundry/issues/7624
- Breaking changes in artifacts / broadcasts
- Additional low-medium impact cheatcodes additions
- Additional low-medium impact commands in `forge`, `cast`, `anvil`, `chisel`
- Non-critical performance optimizations (outside of https://github.com/foundry-rs/foundry/labels/A-testing)

------

Before release:

- [x] https://github.com/foundry-rs/foundry/issues/9419

------

Before publishing:

- Deprecation removal:
  - https://github.com/foundry-rs/foundry/issues/8931

- https://github.com/foundry-rs/foundry/labels/A-releases:
  - https://github.com/foundry-rs/foundry/issues/3895
  - https://github.com/foundry-rs/foundry/issues/494
  - https://github.com/foundry-rs/foundry/issues/2212
  - https://github.com/foundry-rs/foundry/issues/2400
  - https://github.com/foundry-rs/foundry/issues/7044
  - https://github.com/foundry-rs/foundry/issues/5722 
  - https://github.com/foundry-rs/foundry/issues/8671
  - https://github.com/foundry-rs/foundry/issues/7290
  - https://github.com/foundry-rs/foundry/issues/5133
